### ISSUE TITLE
`match_single_binding`: missing brackets

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang/rust-clippy#15018]
### Summary

The fix in #15017 is incomplete and only handles the particular issue that was reported. The problem is that brackets are needed in the general case to create a scope which will restrict the lifetime of created values.

### Reproducer

I tried this code:

```rust
fn main() {
    let a = 10;
    match 11 {
        a => { println!("a = {a}"); }
    }
    println!("a = {a}");
}
```

Before autofix, this code prints:

```txt
a = 11
a = 10
```

After autofix, the code prints:

```txt
a = 11
a = 11
```

because the autofix, even with #15017 applied, reads:

```rust
fn main() {
    let a = 10;
    let a = 11;
    { println!("a = {a}"); }
    println!("a = {a}");
}
```

### Version

```text
Clippy master as of 9e7782b8
```

### Additional Labels

@rustbot label +I-suggestion-causes-bug

**Comment by antonilol** [comment]
I think this is similar, will this be fixed too by the linked pull request?

Code:
```rust
fn what(a: usize, b: usize, c: usize) -> bool {
    a < b
        && match b {
            b => b < c,
        }
}
```

Clippy output:
```
warning: this match could be written as a `let` statement
 --> src/lib.rs:3:12
  |
3 |           && match b {
  |  ____________^
4 | |             b => b < c,
5 | |         }
  | |_________^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
  = note: `#[warn(clippy::match_single_binding)]` on by default
help: consider using a `let` statement
  |
3 ~         && let b = b;
4 +         b < c
  |

warning: `a` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.02s
```

If I apply the fix using rust-analyzer in my IDE i get this expression in the function, which is a syntax error:
```rust
    a < b
        && let b = b;
        b < c
```

Version: clippy 0.1.88 (6b00bc3880 2025-06-23)
